grakn-graph/src/main/java/ai/grakn/graph/internal/ElementFactory.java

```diff
@@ -57,53 +55,61 @@
     }
 
     // ------------------------------------------- Building Castings  --------------------------------------------------
-    CastingImpl buildCasting(Vertex v, Optional<RoleType> type){
+    CastingImpl buildCasting(Vertex v, RoleType type){
         return trackConcept(new CastingImpl(graknGraph, v, type));
     }
 
     // ---------------------------------------- Building Resource Types  -----------------------------------------------
-    <V> ResourceTypeImpl<V> buildResourceType(Vertex v, Optional<ResourceType<V>> type, Optional<ResourceType.DataType<V>> dataType, Optional<Boolean> isUnique){
+    <V> ResourceTypeImpl<V> buildResourceType(Vertex v, ResourceType<V> type, ResourceType.DataType<V> dataType, Boolean isUnique){
         return trackConcept(new ResourceTypeImpl<>(graknGraph, v, type, dataType, isUnique));
     }
 
     // ------------------------------------------ Building Resources
-    <V> ResourceImpl <V> buildResource(Vertex v, Optional<ResourceType<V>> type, Optional<V> value){
+    <V> ResourceImpl <V> buildResource(Vertex v, ResourceType<V> type, V value){
         return trackConcept(new ResourceImpl<>(graknGraph, v, type, value));
     }
 
     // ---------------------------------------- Building Relation Types  -----------------------------------------------
-    RelationTypeImpl buildRelationType(Vertex v, Optional<RelationType> type, Optional<Boolean> isImplicit){
-        return trackConcept(new RelationTypeImpl(graknGraph, v, type, isImplicit));
+    RelationTypeImpl buildRelationType(Vertex v, RelationType type, Boolean isImplicit){
+        if(isImplicit) {
+            return trackConcept(new RelationTypeImpl(graknGraph, v, type, true));
+        } else {
+            return trackConcept(new RelationTypeImpl(graknGraph, v, type)); //No need to save something as non-implicit.
+        }
```

Comment:
No. Because then you get the following fun problem:

    entityType.hasResource(name); //Creates implicit role type and relation type with implicit set to true.
    graph.putRoleType(has-name-owner); //Tries to set the implicit to false (because the constructor demands it and then fails because that property is immutable.  

I can work around this by saying that for the property implicit you cannot change it once it set but then I am starting to put edge cases in the property method which makes it odd.

This sucks, but so do implicit properties so this is the least bad solution.